enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -0,0 +1,209 @@
+`dependencies` field applies to all policy templates in the policy, while
+the template-level `extraDependencies` field can be used to set up
+dependencies for a specific policy template. Like the policy namespace
+selector, the top-level `dependencies` will take priority over the 
+template-level `extraDependencies`, so policy templates that have
+`extraDependencies` satisfied but not `dependencies` will still be marked
+as `pending`.
+
+```
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: my-complicated-policy
+spec:
+  dependencies: //for all policy templates
+    - kind: Policy
+      name: namespace-foo-setup-policy
+      compliance: Compliant
+  policy-templates:
+    - extraDependencies: //for this policy template only
+        - kind: ConfigurationPolicy
+          name: bar
+          compliance: NonCompliant
+      objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: fix-bar-in-foo
+        spec:
+          remediationAction: enforce
+          namespaceSelector:
+            include: ["foo"]
+          object-templates:
+            ...
+```
+
+When the dependencies for a policy template to be evaluated are not met, we plan to add logic
+in the template sync portion of the policy framework addon that will ignore those templates
+until they are ready. Putting this logic in the template sync ensures that whether a policy
+template is evaluated will be determined on the managed cluster level rather than the hub
+level, which is ideal because policy templates on different managed clusters may need to be
+processed differently based on the state of each managed cluster.
+
+If a policy template meets its dependencies and gets processed properly, but at some point
+afterward the state of the cluster changes and those dependencies are no longer met, we do
+not want the template to be active anymore. Utilizing the watch library should allow us to
+follow the dependencies for each template and put the template back into a pending state
+when this happens. We might want to update the policy object deletion logic in the
+configuration policy controller to prune any deletable objects in configuration policies that
+are in a pending state, so that if a dependency is no longer satisfied, any objects created
+on the cluster as a result of that dependency will be deleted.
+
+### Open Questions
+
+How will the "pending" status for policy templates that have not had their conditions met work?
+We know we want to set the compliance of the policy template object to "pending", but different
+policy templates have differently arranged status fields, so there is not a standard field
+to add a message saying why the evaluation of the policy is pending. We also need to investigate
+if this status will have any issues showing up in the hub cluster templates, and how it will look
```

Comment:
Will the status sync be able to flow the pending status to the hub?  Or is there no status for pending policies which today shows up as the yellow triangle.

Reply:
@gparvin I think the pending status would make it to the hub, since we will be setting it on the replicated policies in the managed cluster. However, I think it might be best to still show the yellow triangle - that icon means the policy hasn't been processed on a cluster yet, which is pretty much true if some of its templates are in a pending state. @JustinKuli what do you think?

Reply:
I agree. It shouldn't be a problem for status sync to tell the hub that the policy is pending, and it should provide details on what it's waiting on.

I also agree that the yellow triangle icon would fit here, and we will need to add UI logic to ensure that is what's used while the policy is pending.